### PR TITLE
Tests for shopping cart

### PR DESCRIPTION
**Pull request by BohdanTrue**

### CODE REVIEW

**Review by annton-shakur — APPROVED** [review]
good job!

File: src/test/java/mate/academy/bookstore/controller/CategoryControllerTest.java [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+                technicalCategory,
+                motivationalCategory,
+                mysticCategory);
+
+        CategoryRequestDto[] actual = objectMapper.readValue(
+                result.getResponse().getContentAsByteArray(),
+                CategoryRequestDto[].class
+        );
+
+        EqualsBuilder.reflectionEquals(expected, actual);
```
Comment (yevheniisynytsia) [inline src/test/java/mate/academy/bookstore/controller/CategoryControllerTest.java]: does it work fine when comparing list and array? looks good if so

File: src/test/java/mate/academy/bookstore/controller/ShoppingCartControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+//            );
+//        }
+//    }
+//
+//    @AfterAll
+//    static void afterAll(
+//            @Autowired DataSource dataSource
+//    ) {
+//        teardown(dataSource);
+//    }
```
Comment (yevheniisynytsia) [inline src/test/java/mate/academy/bookstore/controller/ShoppingCartControllerTest.java]: why is it commented?

File: src/test/java/mate/academy/bookstore/repository/book/BookRepositoryTest.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    @Sql(scripts = {
+            "classpath:database/books_categories/delete-books_categories.sql",
+            "classpath:database/categories/delete-categories.sql",
+            "classpath:database/books/delete-books.sql"
+    }, executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD)
+    void findAllBooks_WithValidCategoryId_ReturnOk() {
+        List<Book> actual = bookRepository.findAllByCategoryId(1L);
+
+        assertEquals(1, actual.size());
+        assertEquals("Robert Martin", actual.get(0).getAuthor());
```
Comment (yevheniisynytsia) [inline src/test/java/mate/academy/bookstore/repository/book/BookRepositoryTest.java]: compare the whole object with an expected one to be sure that it is ok completely

File: src/test/java/mate/academy/bookstore/service/impl/BookServiceImplTest.java [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+
+@ExtendWith(MockitoExtension.class)
+class BookServiceImplTest {
+    @InjectMocks
+    private BookServiceImpl bookService;
```
Comment (yevheniisynytsia) [inline src/test/java/mate/academy/bookstore/service/impl/BookServiceImplTest.java]: place it under mocks

File: src/test/java/mate/academy/bookstore/service/impl/BookServiceImplTest.java [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+    @Mock
+    private BookRepository bookRepository;
+
+    @Mock
+    private BookMapper bookMapper;
+
+    @Test
```
Comment (yevheniisynytsia) [inline src/test/java/mate/academy/bookstore/service/impl/BookServiceImplTest.java]: display name

**Review by yevheniisynytsia — CHANGES_REQUESTED** [review]
Great job!
Please consider applying comments to other files

File: src/test/java/mate/academy/bookstore/controller/CategoryControllerTest.java (hunk 1 above)
Comment (BohdanTrue) [reply src/test/java/mate/academy/bookstore/controller/CategoryControllerTest.java]: Yeah, this method can also be used to compare lists or arrays that contain objects with corresponding getters and fields

File: src/test/java/mate/academy/bookstore/repository/book/BookRepositoryTest.java (hunk 3 above)
Comment (BohdanTrue) [reply src/test/java/mate/academy/bookstore/repository/book/BookRepositoryTest.java]: It doesn't refer to this HW, I already did it in my previous HW :D

**Comment by BohdanTrue** [comment]
image
I have this problem which occurs on line 167 in my ShoppingCartControllerTest

**Review by yevheniisynytsia — APPROVED** [review]

**Pull request merged**